pymc3/distributions/multivariate.py

```diff
@@ -557,92 +590,26 @@ class DirichletMultinomial(Discrete):
         Describes shape of distribution. For example if n=array([5, 10]), and
         a=array([1, 1, 1]), shape should be (2, 3).
     """
+    rv_op = dirichlet_multinomial
 
-    def __init__(self, n, a, shape, *args, **kwargs):
-
-        super().__init__(shape=shape, defaults=("_defaultval",), *args, **kwargs)
-
+    @classmethod
+    def dist(cls, n, a, shape, *args, **kwargs):
```

Comment:
We have to be careful with shape. A user can define `size` or `shape` or `dims`, and the rv_ops should be defined in terms of `size` alone.  Therefore, any broadcasting operations should rely on `size` and be contained (if needed) inside the `rng_fn` and `logp` methods.

Reply:
This other WIP PR might provide some clarification: https://github.com/pymc-devs/pymc3/pull/4731/files

Reply:
There are also some broadcast utils in here: https://github.com/aesara-devs/aesara/blob/main/aesara/tensor/random/utils.py

Reply:
Yeah that didn't work anyway -- can't instantiate an RV with `shape`. I got rid of all this shape handling, to see how this goes for now 🤷‍♂️